x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/ReplaceAggregateAggExpressionWithEval.java

```diff
@@ -51,6 +54,14 @@ protected LogicalPlan rule(Aggregate aggregate) {
         AttributeMap<Expression> aliases = new AttributeMap<>();
         aggregate.forEachExpressionUp(Alias.class, a -> aliases.put(a.toAttribute(), a.child()));
 
+        // build grouping functions map
+        Map<GroupingFunction, Attribute> groupingAttributes = new HashMap<>();
```

Comment:
Can we be specific for `Categorize`, instead of applying this code to all grouping functions?

The added code in this file seems correct to me, but it has a chance to introduce edge cases to `BUCKET` that we simply have no tests against.

And even if it doesn't, it means it doesn't have an effect on `BUCKET` and is thus needlessly general because I can remove it without consequences for `BUCKET` but that's absolutely not evident.

Reply:
Sure. I changed the types to Categorize, and added a comment explaining why Bucket doesn't apply. Eventually, this should be change to a `StatefulGroupingFunction` or whatever we create, if we do so. With all the other usages of `Categorize` around the rules